Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -127,10 +148,134 @@ public void Show(DataTemplate dataTemplate, int duration = 0)
         /// </summary>
         public void Dismiss()
         {
+            Dismiss(InAppNotificationDismissKind.Programmatic);
+        }
+
+        /// <summary>
+        /// Dismiss the notification
+        /// </summary>
+        /// <param name="dismissKind">Kind of action that triggered dismiss event</param>
+        private void Dismiss(InAppNotificationDismissKind dismissKind)
+        {
             if (Visibility == Visibility.Visible)
             {
+                _animationTimer.Stop();
+
+                var eventArgs = new InAppNotificationDismissingEventArgs(dismissKind);
+                Dismissing?.Invoke(this, eventArgs);
+
+                if (eventArgs.Cancel)
+                {
+                    return;
+                }
+
                 VisualStateManager.GoToState(this, StateContentCollapsed, true);
-                Dismissed?.Invoke(this, EventArgs.Empty);
+
+                _animationTimer.Interval = TimeSpan.FromMilliseconds(AnimationDuration);
+                _animationTimer.Tick += DismissAnimationTimer_Tick;
+                _animationTimer.Start();
+            }
+        }
+
+        private void UpdateAnimationDuration(int duration)
+        {
+            if (_visualStateGroup != null)
+            {
+                var keyTimeFromAnimationDuration = KeyTime.FromTimeSpan(TimeSpan.FromMilliseconds(duration));
+
+                foreach (var state in _visualStateGroup.States)
+                {
+                    foreach (var timeline in state.Storyboard.Children)
+                    {
+                        if (timeline is DoubleAnimationUsingKeyFrames daukf)
+                        {
+                            var keyFramesCount = daukf.KeyFrames.Count;
+                            if (keyFramesCount > 1)
+                            {
+                                daukf.KeyFrames[keyFramesCount - 1].KeyTime = keyTimeFromAnimationDuration;
+                            }
+                        }
+
+                        if (timeline is ObjectAnimationUsingKeyFrames oaukf)
+                        {
+                            var keyFramesCount = oaukf.KeyFrames.Count;
+                            if (keyFramesCount > 1)
+                            {
+                                oaukf.KeyFrames[keyFramesCount - 1].KeyTime = keyTimeFromAnimationDuration;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private void UpdateVerticalOffset(double verticalOffset)
+        {
```

Comment:
This is also seems like it will be messing with my custom animations? Instead of doing the Update(Animation/Offset) through code, is it possible to instead do this through TemplateBinding in the template itself?